aws/resource_aws_mwaa_environment.go

```diff
@@ -0,0 +1,652 @@
+		ExecutionRoleArn:     aws.String(d.Get("execution_role_arn").(string)),
+		Name:                 aws.String(d.Get("name").(string)),
+		NetworkConfiguration: expandMwaaEnvironmentNetworkConfigurationCreate(d.Get("network_configuration").([]interface{})),
+		SourceBucketArn:      aws.String(d.Get("source_bucket_arn").(string)),
+	}
+
+	if v, ok := d.GetOk("airflow_configuration_options"); ok {
+		input.AirflowConfigurationOptions = stringMapToPointers(v.(map[string]interface{}))
+	}
+
+	if v, ok := d.GetOk("airflow_version"); ok {
+		input.AirflowVersion = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("environment_class"); ok {
+		input.EnvironmentClass = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("kms_key"); ok {
+		input.KmsKey = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("logging_configuration"); ok {
+		input.LoggingConfiguration = expandMwaaEnvironmentLoggingConfiguration(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("max_workers"); ok {
+		input.MaxWorkers = aws.Int64(int64(v.(int)))
+	}
+
+	if v, ok := d.GetOk("plugins_s3_object_version"); ok {
+		input.PluginsS3ObjectVersion = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("plugins_s3_path"); ok {
+		input.PluginsS3Path = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("requirements_s3_object_version"); ok {
+		input.RequirementsS3ObjectVersion = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("requirements_s3_path"); ok {
+		input.RequirementsS3Path = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("webserver_access_mode"); ok {
+		input.WebserverAccessMode = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("weekly_maintenance_window_start"); ok {
+		input.WeeklyMaintenanceWindowStart = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("tags"); ok {
+		input.Tags = keyvaluetags.New(v.(map[string]interface{})).IgnoreAws().MwaaTags()
+	}
+
+	log.Printf("[INFO] Creating MWAA Environment: %s", input)
+	_, err := conn.CreateEnvironment(&input)
+	if err != nil {
+		return fmt.Errorf("error creating MWAA Environment: %w", err)
+	}
+
+	d.SetId(aws.StringValue(input.Name))
+
+	if _, err := waiter.EnvironmentCreated(conn, d.Id()); err != nil {
+		return fmt.Errorf("error waiting for MWAA Environment (%s) creation: %w", d.Id(), err)
+	}
+
+	return resourceAwsMwaaEnvironmentRead(d, meta)
+}
+
+func resourceAwsMwaaEnvironmentRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).mwaaconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	log.Printf("[INFO] Reading MWAA Environment: %s", d.Id())
+
+	environment, err := finder.EnvironmentByName(conn, d.Id())
+
+	if err != nil {
+		if isAWSErr(err, mwaa.ErrCodeResourceNotFoundException, "") && !d.IsNewResource() {
+			log.Printf("[WARN] MWAA Environment %q not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("error reading MWAA Environment (%s): %w", d.Id(), err)
+	}
+
```

Comment:
Nit: To prevent potential panics, it is generally good to perform a final `nil` check before accessing the object:

```suggestion

	if environment == nil {
		return fmt.Errorf("error reading MWAA Environment (%s): empty response", d.Id())
	}

```